### PR TITLE
Tools #2 git, vim, compilers

### PR DESCRIPTION
**Pull request by NeoBauer**

### CODE REVIEW

File: src/functions.cpp [hunk 1]
```diff
@@ -1,9 +1,9 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include <iostream>
-#include <limits>
+#include <algorithm>
 
 void print(std::vector<int> const & numbers) {
-    for (int i = 0; i < numbers.size(); ++i) {
+    for (unsigned int i = 0; i < numbers.size(); ++i) {
```
Comment (DamiXioR) [inline src/functions.cpp:6]: use std::size_t with arrays etc.

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 -Wall -Wextra -pedantic -Wconversion -Werror -Iinc/ src/*.cpp  -o main
```
Comment (DamiXioR) [inline Makefile:2]: -I./inc looks better

File: src/functions.cpp [hunk 3]
```diff
@@ -1,9 +1,9 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (DamiXioR) [inline src/functions.cpp:1]: unnecessary comments

File: src/main.cpp [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #include <vector>
 #include <iostream>
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (DamiXioR) [inline src/main.cpp:3]: unnecessary comments

**Review by DamiXioR — COMMENTED** [review]
have a nice day!

File: src/functions.cpp [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 #include <iostream>
-#include <limits>
+#include <algorithm>
```
Comment (Marcinati) [inline src/functions.cpp:3]: Sort headers.

**Review by Marcinati — CHANGES_REQUESTED** [review]

File: tutor.txt [hunk 6]
```diff
@@ -0,0 +1,976 @@
...
+  3. Now move to the fourth line.
+  4. Type   2dd   to delete two lines.
+
+--->  1)  Roses are red,
+--->  2)  Mud is fun,
+--->  3)  Violets are blue,
+--->  4)  I have a car,
+--->  5)  Clocks tell time,
+--->  6)  Sugar is sweet
+--->  7)  And so are you.
```
Comment (ziobron) [inline tutor.txt:320]: not completed

File: tutor.txt [hunk 7]
```diff
@@ -0,0 +1,976 @@
...
+
+  1. Move the cursor to the line below marked --->.
+
+  2. Type  :s/thee/the <ENTER> .  Note that this command only changes the
+     first occurrence of "thee" in the line.
+
+  3. Now type  :s/thee/the/g .  Adding the  g  flag means to substitute
+     globally in the line, change all occurrences of "thee" in the line.
+
+---> thee best time to see thee flowers is in thee spring.
```
Comment (ziobron) [inline tutor.txt:567]: not completed

File: Makefile [hunk 8]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 -Iinc/ src/*.cpp  -o main
```
Comment (ziobron) [inline Makefile]: it should be done after merging

**Comment by ziobron** [comment]
Anyway - max points
+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }

**Pull request closed without merging**